### DISCUSSION TITLE
Does the jar build support java 6?

### DISCUSSION THREAD

**Discussion by zukarusan** [JSQLParser/JSqlParser#1787] [Q&A]
Hi All,
I think looking at JSqlParser is perfect as a SQL parsing solution in Java. But I am currently doing a legacy project in my workspace. It is required me to have Java 6. Does this support?

Thanks

**Comment by manticore-projects** [comment]
Greetings.

Unusual request, but I was able to build the Library for Java 1.6 with

```xml
java {
    sourceCompatibility = '1.8'
    targetCompatibility = '1.6'
}
```

Please try the attached JAR file, it should be Java 1.6 compatible -- but I can't try that myself because I was not born yet when 1.6 was available :-D So if it does not work, then you will be on your own.
[JSQLParser-4.7-SNAPSHOT-Java6.zip](https://github.com/JSQLParser/JSqlParser/files/11474347/JSQLParser-4.7-SNAPSHOT-Java6.zip)

**Comment by wumpz** [comment]
Since it uses lambdas and streaming the target compatibility is not enough. You are not able to build a version that uses JDK 1.6 API without refactoring. Version 1.2 switched to JDK 1.7. So this should be useable with Jdk 1.6.  V3 made the switch to JDK 1.8.
